InterfacePython/testBasicIO.py

```diff
@@ -410,6 +410,59 @@ def checkReadWriteMisc(self):
                 gridB.read("testSave")
                 ttc.compareGrids(gridA, gridB)
 
+    def checkReadWriteCustomTabulated(self):
+        '''
+        Test reading and writing of the CustomTabulated class.
+        '''
+        rng = np.random.default_rng(777);
+        description = "testCT"
+        def create_nodes(j, in_rng):
+            nodes = in_rng.random(j) * 2.0 - 1.0
+            nodes.sort()
+            return nodes
+        def create_weights(j, in_rng):
+            weights = in_rng.random(j)
+            weights = 2.0 * weights / weights.sum()
+            return weights
+        # Read and write from explicitly given data.
+        for i in range(4):
+            num_levels = i
+            num_nodes = np.array([3*(j+1) for j in range(i)])
+            precision = np.array([2*(j+1)-1 for j in range(i)])
+            nodes = [create_nodes(j, rng) for j in num_nodes]
+            weights = [create_weights(j, rng) for j in num_nodes]
+            ctA = TasmanianSG.makeCustomTabulatedFromData(num_levels, num_nodes, precision, nodes, weights, description)
+            ctB = TasmanianSG.CustomTabulated()
+            ctA.write("testSave")
+            ctB.read("testSave")
+            ttc.compareCustomTabulated(ctA, ctB)
```

Comment:
This is effectively the same test as the one below, i.e., makes sure that the data that you save matches the data that you read. However, you don't explicitly test if the loaded points and weights match what you set in the random vectors.

The read/write test is OK, but you have to check the loaded points and weights.

Reply:
Added code to test if the read `CustomTabulated` instance has the right loaded weights and nodes.